### PR TITLE
chore: Update DataFusion + arrow/parquet/arrow-flight to 47.0.0

### PR DESCRIPTION
**Pull request by alamb**
Closes https://github.com/influxdata/idpe/issues/18117
Closes https://github.com/influxdata/idpe/issues/18105

# Rationale
* Keep up with dependencies
* get https://github.com/apache/arrow-rs/pull/3558, to help with memory use
* get https://github.com/apache/arrow-datafusion/pull/7614 to help with timezones

# Changes
* Update DataFusion pin to https://github.com/apache/arrow-datafusion/commit/b1d134e9ff37724459cb5090a6346a85152a1db7
* Update arrow/parquet/arrow-flight to `47.0.0`
* Update tonic (and related crates) to `0.10`
* Update prost (and related crates) to `0.12`
* Update pbjson to `0.6`

# Related
-  [ ] https://github.com/tikv/pprof-rs/pull/223

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -4535,15 +4563,37 @@ dependencies = [
...
+ "prost 0.11.9",
+ "prost-types 0.11.9",
  "regex",
  "syn 1.0.109",
  "tempfile",
  "which",
 ]
 
+[[package]]
+name = "prost-build"
```
Comment (alamb) [inline Cargo.lock:4576]: pprof still uses prost/prost build `11.0`, so until that is updated IOx will compile two versions of the stack. I have a PR upstream to update prost: https://github.com/tikv/pprof-rs/pull/223

File: authz/src/permission.rs [hunk 2]
```diff
@@ -75,13 +75,13 @@ impl TryFrom<proto::Permission> for Permission {
         match value.permission_one_of {
             Some(proto::permission::PermissionOneOf::ResourceAction(ra)) => {
                 let r = Resource::try_from_proto(
-                    proto::resource_action_permission::ResourceType::from_i32(ra.resource_type)
-                        .ok_or(IncompatiblePermissionError {})?,
+                    proto::resource_action_permission::ResourceType::try_from(ra.resource_type)
```
Comment (alamb) [inline authz/src/permission.rs:78]: `from_i32` has been deprecated in favor of `try_from`

File: arrow_util/Cargo.toml [hunk 3]
```diff
@@ -8,8 +8,7 @@ license.workspace = true
 
 [dependencies]
 ahash = { version = "0.8", default-features = false, features = ["runtime-rng"] }
-# need dyn_cmp_dict feature for comparing dictionary arrays
```
Comment (alamb) [inline arrow_util/Cargo.toml:11]: This feature flag has been removed from arrow -- see https://github.com/apache/arrow-rs/pull/4766

File: Cargo.lock (hunk 1 above)
Comment (alamb) [reply Cargo.lock:4576]: It also looks like `tokio-console` uses an older version of tonic -- they have updated it upstream -- https://github.com/tokio-rs/console/pull/464 but it does not appear to be available in any release on crates.io yet

**Review by tustvold — APPROVED** [review]

**Comment by alamb** [comment]
I plan to merge this tomorrow when I have sufficient time to observe the rollout and its effect on production

**Pull request merged**